frontend/packages/component-library/src/cms/snapshot/Snapshot.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import {Key, HTMLAttributes, ReactNode} from 'react';
+
+import FontAwesomeV6Icon, {FontAwesomeV6IconProps} from '@/fontAwesomeV6Icon';
+
+import moduleStyles from './snapshot.module.scss';
+
+export type SnapshotItem = {
+  key: Key;
+  icon: FontAwesomeV6IconProps;
+  label: string | ReactNode;
+  content: string | ReactNode;
+};
+
+export interface SnapshotProps extends HTMLAttributes<HTMLDListElement> {
+  /** Snapshot items */
+  items: SnapshotItem[];
+  /** Snapshot class */
+  className?: string;
+}
+
+/**
+ * ## Production-ready Checklist:
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
+ *  * (see ./__tests__/Snapshot.test.tsx)
+ *  * (✔) passes accessibility checks;
+ *
+ * ### Status: ```Ready for dev```
+ *
+ * Design System: Snapshot Component.
+ * Acts as a container for snapshot details.
+ */
+const Snapshot: React.FC<SnapshotProps> = ({
+  items,
+  className,
+  ...HTMLAttributes
+}: SnapshotProps) => (
+  <dl
+    className={classNames(moduleStyles.snapshot, className)}
+    {...HTMLAttributes}
+  >
+    {items.map(({key, icon, label, content}) => (
+      <div key={key} className={moduleStyles.snapshotItem}>
+        <FontAwesomeV6Icon
+          {...icon}
+          aria-hidden="true"
+          className={classNames(moduleStyles.snapshotItemIcon, icon.className)}
+        />
+
+        <dt className={moduleStyles.snapshotItemLabel}>
+          {label}
```

Comment:
It might be possible for these to be undefined so it would be good to add some defensive code to the children of items

Reply:
This is a required prop, so we either don't need to do that, or we need to set it to be optional